eng/pipelines/templates/scripts/validate-changes-on-allowed-path.ps1

```diff
@@ -0,0 +1,64 @@
+<#
+.SYNOPSIS
+
+.DESCRIPTION
+The script is to validate if the diff changes between base and target are within the scope of allowlist.
+
+.PARAMETER BaseBranch
+The base branch where we want to merge changes from. E.g. main branch in docs repo
+
+.PARAMETER TargetBranch
+The target branch where we want to merge changes to. E.g. live branch in docs repo
+
+.PARAMETER AllowlistPath
+The file path of allowlist.
+#>
+param(
+  [Parameter(mandatory=$true)]
+  [string] $BaseBranch,
+  [Parameter(mandatory=$true)]
+  [string] $TargetBranch,
+  [Parameter(mandatory=$true)]
+  [string] $AllowlistPath
+)
+
+# Fetch all branches, e.g. main, live
+Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" fetch origin"
+git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" fetch origin | Write-Host
+
+Write-Host "Check existence on target branch"
+git rev-parse --verify $TargetBranch 2>$null
+if ($LASTEXITCODE -ne 0) {
+  Write-Host "Creating target branch $TargetBranch..."
+  git checkout -b $TargetBranch -t origin/$TargetBranch | Write-Host
+}
+
+# Working branch is base branch, e.g. main
+Write-Host "Check existence on base branch"
+git rev-parse --verify $BaseBranch 2>$null
+if ($LASTEXITCODE -eq 0) {
+  Write-Host "Swtiching to base branch $BaseBranch..."
+  git checkout $BaseBranch | Write-Host
+}
+else {
+  Write-Host "Creating base branch $BaseBranch..."
+  git checkout -b $BaseBranch -t origin/$BaseBranch | Write-Host
+}
+
+# git diff on all files
+$changedFiles = git diff --name-only $TargetBranch
+Write-Host "Here is the total number of the changed files: $changedFiles.Count"
+
+# Get all allowPath, allowPath is the regex of the path
+$allowedPath = Get-Content -path $AllowlistPath
+
+# forloop to see whether there are any changes outside of the allowlist
+Write-Host "Validating the changed files..."
+foreach ($file in $changedFiles) {
+    $matched = $allowedPath | Where-Object {$file -match $_}
+    if ($matched.Count -eq 0) {
+        Write-Error "We DO NOT allow to check in the changes outside of the allowlist automaticaly. `r`nPlease manually merge $file to $TargetBranch."
+        exit 1
```

Comment:
We should try to capture a list of files that changed instead of just the first one.

Reply:
Will do